utilities/ttl/db_ttl_impl.h

```diff
@@ -107,15 +108,27 @@ class DBWithTTLImpl : public DBWithTTL {
  private:
   // remember whether the Close completes or not
   bool closed_;
+  bool strict_;
+  std::unordered_map<std::string, int32_t> ttls_;
+  SystemClock* clock_;
 };
 
 class TtlIterator : public Iterator {
  public:
-  explicit TtlIterator(Iterator* iter) : iter_(iter) { assert(iter_); }
+  explicit TtlIterator(Iterator* iter, bool strict, int32_t ttl, SystemClock* clock) :
+    iter_(iter), strict_(strict), ttl_(ttl), clock_(clock) { assert(iter_); }
 
   ~TtlIterator() { delete iter_; }
 
-  bool Valid() const override { return iter_->Valid(); }
+  bool Valid() const override {
+    if (!iter_->Valid()) {
+      return false;
+    }
+    if (strict_) {
+      return !DBWithTTLImpl::IsStale(iter_->value(), ttl_, clock_);
```

Comment:
@ltamasi pointed out an issue with this internally. `!Valid()` means the iterator is invalid, that is, it either encountered an error or there are no more keys for it to return. The case of one key being stale meets neither of those criteria. For example, if the operation was `Next()`, we should advance past stale keys to try to reach a non-stale key.